tests/statespace_test.h

```diff
@@ -0,0 +1,433 @@
+// Copyright 2019 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STATESPACE_TEST_H_
+#define STATESPACE_TEST_H_
+
+#include <array>
+#include <cmath>
+#include <complex>
+#include <cstdint>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "../lib/circuit_reader.h"
+#include "../lib/fuser_basic.h"
+#include "../lib/gates_def.h"
+#include "../lib/io.h"
+#include "../lib/run_qsim.h"
+
+namespace qsim {
+
+template <typename StateSpace>
+void TestSamplingBasic() {
```

Comment:
Minor nit: `TestSamplingBasic` suggests to me that this is for `StateSpaceBasic` only - maybe `TestSamplingSmall`?